src/modals/MarkModal.tsx

```diff
@@ -1,127 +1,83 @@
-                            );
-                        } else {
-                            cycles.unshift(
-                                {
-                                    cycleLength: 28,
-                                    periodLength: props.period,
-                                    startDate: props.date,
-                                }
-                            );
-                        }
-
-                        Promise.all([
-                            updateCycles([...cycles])
-                        ]).then(() => {
-                            console.log("All new values are set, setIsOpen(false)");
-                            props.closeModal();
-                        }).catch((err) => console.error(err));
-                    }
-                }}
-            >
-                Ok
-            </IonButton>
-            <IonButton
-                style={{ float: "right" }}
-                color="dark-basic"
-                fill="clear"
-                onClick={() => {
-                    props.closeModal();
-                    props.setDate("");
-                }}>
-                Cancel
-            </IonButton>
-        </>
-    );
-};
+import { format } from 'date-fns'
 
 interface PropsMarkModal {
     isOpen: boolean;
     setIsOpen: (newIsOpen: boolean) => void;
 }
 
 const MarkModal = (props: PropsMarkModal) => {
-    const [date, setDate] = useState("");
-    const [period, setPeriod] = useState(0);
-
-    const closeModal = () => {
-        props.setIsOpen(false);
-        setDate("");
-    };
-
-    const selectOptions = {
-        cssClass: "mark-select-header",
+    const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
+    const { cycles, updateCycles } = useContext(CyclesContext);
+    const lengthOfPeriod = useAverageLengthOfPeriod();
+    let markPeriodDays: string[] = [];
+
+    const nowDate = new Date();
+    nowDate.setHours(0, 0, 0, 0);
+
+    const nextCycleFinish: Date = new Date();
+    nextCycleFinish.setDate(nowDate.getDate() + lengthOfPeriod);
+    nextCycleFinish.setHours(0, 0, 0, 0);
+
+    function isLastPeriodDay(date: Date) {
+        date.setHours(0, 0, 0, 0);
+
+        for (let i = 0; i < cycles.length; ++i) {
+            const cycleStart: Date = new Date(cycles[i].startDate);
+            const cycleFinish: Date = new Date(cycles[i].startDate);
+            cycleFinish.setDate(cycleFinish.getDate() + cycles[i].periodLength);
+
+            if (date >= cycleStart && date < cycleFinish) {
+                return true;
+            }
+        };
+
+        return false;
+    }
+
+    function nextPeriodDays(): string[] {
+        let periodDates = [format(nowDate, 'yyyy-MM-dd')];
+        let lengthOfNextPeriod = lengthOfPeriod;
+        if (lengthOfNextPeriod === 0) {
+            const averageLengthOfPeriod = 5;
+            lengthOfNextPeriod = averageLengthOfPeriod;
+        }
+
+        for (let i = 1; i < lengthOfNextPeriod; ++i) {
```

Comment:
You can rewrite this function to something like that:

```js
const periodDates: string[] = [];

for (let day = 0; day < lengthOfPeriod || 5; day++) {
  const periodDay = new Date(nowDate);
  periodDay.setHours(0, 0, 0, 0);
  periodDay.setDate(periodDay.getDate() + day);

  periodDates.push(format(periodDay, "yyyy-MM-dd");
}

return periodDates;
```

Reply:
Fixed